### PR TITLE
Add an API to verify sessions

### PR DESCRIPTION
**Pull request by grepsedawk**
Depends on #9

### CODE REVIEW

**Comment by ekate** [comment]
@jgpawletko it does what I expect it to do so if you are o.k. with it I can merge it

File: app/models/session.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  def generate_token
+    self.token = loop do
+      random_token = SecureRandom.urlsafe_base64(nil, false)
+      break random_token unless Session.exists?(token: random_token)
+    end
+  end
+
+  def set_expiry
+    self.expires_at = 2.seconds.from_now
```
Comment (jgpawletko) [inline app/models/session.rb]: Hi @pachonk ,
did you mean `self.expires_at = 2.hours.from_now`?

**Comment by jgpawletko** [comment]
Hi @ekate,
I just added a question for alex.

File: app/models/session.rb (hunk 1 above)
Comment (grepsedawk) [reply app/models/session.rb]: Ah, good catch! Thanks. Fixed.

**Comment by grepsedawk** [comment]
Resolved. Thanks.

**Comment by grepsedawk** [comment]
This depends on #9 to be merged prior.

**Comment by jgpawletko** [comment]
Hi @ekate,
ok to merge.

**Comment by grepsedawk** [comment]
Test failing after resolving merge conflict (during rebase), will fix it tomorrow

**Pull request merged**